keps/sig-multicluster/2149-clusterid/README.md

```diff
@@ -0,0 +1,705 @@
+a broad sense previously ([see this doc](https://docs.google.com/document/d/1F__vEKeI41P7PPUCMM9PVPYY34pyrvQI5rbTJVnS5c4/edit?usp=sharing), and was scoped 
+down in response to actual observed use cases in the latest community discussion on
+which this KEP is based ([doc](https://docs.google.com/document/d/1S0u6xzP2gcJKPipA6tBNDNuid76nVKeGhTk7PrCIuQY/edit?usp=sharing)). The motivation
+of this KEP is to provide a flexible but useful baseline for clusterID that can
+work with the known use cases.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+* Propose a standard for how cluster identification metadata should be stored and 
+managed as Kubernetes resources
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+* Define any characteristics of the system that tracks cluster IDs within a cluster (i.e. a cluster registry)
+* Solve any problems without specific, tangible use cases (though we will leave room for extension).
+
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### Overview
+Each cluster in a ClusterSet will be assigned a unique identifier that lives at least as long as its membership in the set, and is immutable for its lifetime. This identifier will be stored in a new cluster-scoped `ClusterClaim` CR with the well known name `id.k8s.io` that may be referenced by workloads within the cluster. The identifier must be a valid [RFC-1123](https://tools.ietf.org/html/rfc1123) DNS label, and may be created by an implementation dependent mechanism.
+
+While a member of a ClusterSet, a cluster will also have an additional `clusterset.k8s.io ClusterClaim` which describes its current membership. This claim must be present exactly as long as the set membership lasts, and removed when the cluster is no longer a member.
+
+### User Stories
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### ClusterSet membership
+
+I have some set of clusters working together and need a way to uniquely identify them within the system that I use to track membership.
+
+#### Multi-Cluster Services
+
+I have a headless multi-cluster service deployed across clusters in my ClusterSet with similarly named pods in each cluster. I need a way to disambiguate each backend pod via DNS.
```

Comment:
Maybe not within scope of this kep, but have couple questions about impact on MCS API

* As per MCS API spec, today multicluster services use `clusterset.local` instead of `cluster.local` Would this be replaced with `clusterset.<clustersetId>` ?
* Today MCS API recommends using `<hostname>.<clustername>.<svc>.<ns>.svc.clusterset.local` to address individual pod in a multicluster/clusterset. Would `clustername` be replaced with the new `clusterID`?

Reply:
>As per MCS API spec, today multicluster services use clusterset.local instead of cluster.local Would this be replaced with clusterset.<clustersetId> ?

I don't think we'd want to replace clusterset.local. I've been thinking clustersetid could be used by tools running in the cluster to identify the clusterset it belongs to - not for MCS consumers, but for MCS implementations.

> Today MCS API recommends using <hostname>.<clustername>.<svc>.<ns>.svc.clusterset.local to address individual pod in a multicluster/clusterset. Would clustername be replaced with the new clusterID?

Exactly - clustername in this case would be clusterid. I think we need some examples :)

Reply:
Put a placeholder on this in the doc in 56c8fc1f6afea30975061a06bb0ff6df9dfbe750